### PR TITLE
Create initial BFD state only when a real BFD state shows up

### LINKED ISSUE #197: BFD: bfdix, bfddiscr and bfdaddr attributes not created on all bfd case'es
Running: commit 75f276dd5a8b8d3d96f368032aafd53bf43e3ffb

it seams that there are instances where bfdix, bfddiscr and bfdaddr attributes are not populated to BFD case. 
This makes curitz crash as those attributes are expected.
What's make these failes are for me unknown.

Here are some debug from my running system:
```In [2]: for x in sess.cases_iter():
   ...:     if x.type == caseType.BFD:
   ...:         print(f"Case {x!r} type {x.type}")
   ...:         print(x.keys())
   ...:         if not 'bfdix' in x.keys():
   ...:             pprint(x.__dict__)
   ...:
Case <class 'zinolib.ritz.Case'>(6342) type caseType.BFD
['id', 'router', 'type', 'state', 'opened', 'updated', 'priority', 'polladdr', 'ifindex', 'bfdstate', 'bfdix', 'bfddiscr', 'bfdaddr', 'history', 'log']
Case <class 'zinolib.ritz.Case'>(6347) type caseType.BFD
['id', 'router', 'type', 'state', 'opened', 'updated', 'priority', 'polladdr', 'ifindex', 'bfdstate', 'bfdix', 'bfddiscr', 'bfdaddr', 'history', 'log']
Case <class 'zinolib.ritz.Case'>(6378) type caseType.BFD
['id', 'router', 'type', 'state', 'opened', 'updated', 'priority', 'polladdr', 'ifindex', 'bfdstate', 'bfdix', 'bfddiscr', 'bfdaddr', 'history', 'log']
Case <class 'zinolib.ritz.Case'>(6391) type caseType.BFD
['id', 'router', 'type', 'state', 'opened', 'updated', 'priority', 'polladdr', 'ifindex', 'bfdstate', 'history', 'log']
{'_attrs': {'bfdstate': 'noSession',
            'id': 6391,
            'ifindex': 597,
            'opened': datetime.datetime(2024, 3, 19, 17, 40, 17),
            'polladdr': IPv4Address('x.x.0.166'),
            'priority': 100,
            'router': 'xxxx-gw1',
            'state': <caseState.OPEN: 'open'>,
            'type': <caseType.BFD: 'bfd'>,
            'updated': datetime.datetime(2024, 3, 19, 17, 40, 17)},
 '_caseid': 6391,
 '_zino': <zinolib.ritz.ritz object at 0x7f0bff58e620>}
Case <class 'zinolib.ritz.Case'>(6393) type caseType.BFD
['id', 'router', 'type', 'state', 'opened', 'updated', 'priority', 'polladdr', 'ifindex', 'bfdstate', 'bfdix', 'bfddiscr', 'bfdaddr', 'history', 'log']
Case <class 'zinolib.ritz.Case'>(6418) type caseType.BFD
['id', 'router', 'type', 'state', 'opened', 'updated', 'priority', 'polladdr', 'ifindex', 'bfdstate', 'bfdix', 'bfddiscr', 'bfdaddr', 'history', 'log']
Case <class 'zinolib.ritz.Case'>(6421) type caseType.BFD
['id', 'router', 'type', 'state', 'opened', 'updated', 'priority', 'polladdr', 'ifindex', 'bfdstate', 'bfdix', 'bfddiscr', 'bfdaddr', 'history', 'log']
Case <class 'zinolib.ritz.Case'>(6424) type caseType.BFD
['id', 'router', 'type', 'state', 'opened', 'updated', 'priority', 'polladdr', 'ifindex', 'bfdstate', 'bfdix', 'bfddiscr', 'bfdaddr', 'history', 'log']
Case <class 'zinolib.ritz.Case'>(6451) type caseType.BFD
['id', 'router', 'type', 'state', 'opened', 'updated', 'priority', 'polladdr', 'ifindex', 'bfdstate', 'history', 'log']
{'_attrs': {'bfdstate': 'noSession',
            'id': 6451,
            'ifindex': 585,
            'opened': datetime.datetime(2024, 3, 22, 19, 16, 36),
            'polladdr': IPv4Address('x.x.0.129'),
            'priority': 100,
            'router': 'xxx-gw4',
            'state': <caseState.OPEN: 'open'>,
            'type': <caseType.BFD: 'bfd'>,
            'updated': datetime.datetime(2024, 3, 22, 19, 16, 36)},
 '_caseid': 6451,
 '_zino': <zinolib.ritz.ritz object at 0x7f0bff58e620>}
 ```

I've scrambled sensitive information, i've also taken a backup of the current "state" you want to debug on a "live" system

### PR DESCRIPTION
**Pull request by stveit**
Fixes #197 

Makes initial event creation more similar to how original Zino does it. instead of creating a placeholder noSession state, wait to create a state until something actually shows up. Then, if a state is created for the first time and nothing previous to compare against, create event if state is not up.

Zino1 source: https://gitlab.sikt.no/verktoy/zino/-/blob/master/zino/bfd.tcl?ref_type=heads#L242

This should also solve the issue of missing fields in #197

### CODE REVIEW

File: src/zino/tasks/bfdtask.py [hunk 1]
```diff
@@ -61,6 +61,8 @@ def _update_state(self, port: Port, new_state: BFDState):
         if port.bfd_state:
             if port.bfd_state.session_state != new_state.session_state:
                 self._create_or_update_event(port, new_state)
+        elif new_state.session_state != BFDSessState.UP:
+            self._create_or_update_event(port, new_state)
```
Comment (johannaengland) [inline src/zino/tasks/bfdtask.py:65]: It would be good to also then update the docstring of this function, because as I understand it this contradicts it

**Review by johannaengland — APPROVED** [review]
Just a tiny comment, otherwise looks good!

File: src/zino/tasks/bfdtask.py (hunk 1 above)
Comment (stveit) [reply src/zino/tasks/bfdtask.py:64]: Did you mean the comment right beneath the docstring? Yes that one shouldve been removed, done it now. Docstring itself still seems OK (although a bit vague)

Comment (johannaengland) [reply src/zino/tasks/bfdtask.py:64]: Yes, sorry, I meant the comment :D That's the consequences of reviewing a PR "just before I go home" :grin:

**Review by johannaengland — APPROVED** [review]

**Pull request merged**